web-lib-funcs.pl

```diff
@@ -12175,6 +12180,35 @@ sub webmin_user_is_admin
 return &webmin_user_login_mode() eq 'root';
 }
 
+# webmin_user_is()
+# Returns 1 if currently logged in user belongs to one
+# of the requested types: root, safe-user, mail-user,
+# cloud-owner, virtual-reseller, virtual-reseller
+# Simply a convenience wrapper function
+sub webmin_user_is
+{
+my ($user_type) = @_;
+
+# Is user root/admin
+return &webmin_user_login_mode() eq 'root'
```

Comment:
Better to call webmin_user_login_mode just once, store the result in a variable, and check that below.

Reply:
> Better to call webmin_user_login_mode just once, store the result in a variable, and check that below.

Yes, great suggestion! Have a look at this patch - https://github.com/webmin/webmin/pull/1649/commits/16e8f9ea9d0be6a6e165ba9b7a20edaeda46b3d1

Reply:
No I meant to just call it once in this function. I don't like the use of `state` to do caching in this case.

Reply:
>  I don't like the use of state to do caching in this case.

Why not? Do you expect that mode can change during the call?

Reply:
> Better to call webmin_user_login_mode just once, store the result in a variable, and check that below.

I see now, done - https://github.com/webmin/webmin/pull/1649/commits/5df581e103238dcd691981b01f4896a520ed9f04

> I don't like the use of state to do caching in this case.

This function can be called dozens of times per call. Unless we expect mode to be changed during the call, I would keep caching. It's very effective.

Reply:
I'd just prefer to use an explicit global variable as the cache then, rather than using `state`

Reply:
> Thanks, this looks a lot better

Okay, great then.